src/components/pages/AppMarketplace/components/AppListGroup/index.tsx

```diff
@@ -32,26 +32,34 @@ export const AppListGroup = ({
 }) => {
   const { t } = useTranslation()
   const [itemsShown, setItemsShown] = useState('4')
-
   const itemsToShow = items.slice(0, Number(itemsShown))
 
-  const increaseItemsShown = () => {
-    setItemsShown((prevState) => {
-      return (Number(prevState) + 4).toString()
-    })
+  const handleMoreAndCollapse = () => {
+    if (itemsToShow.length >= items.length) {
+      setItemsShown(() => {
+        return (4).toString()
+      })
+    } else
+      setItemsShown(() => {
+        return items.length.toString()
+      })
   }
```

Comment:
First of all why is `itemsToShow` a string in the old code? That doesn't make any sense and should be fixed here to number. Then the entire function can be written more readable as one line (maybe prettier might reformat):
` const handleMoreAndCollapse = () => setItemsShow(itemsToShow.length >= items.length ? 4 : items.length `